packages/common/src/hardhat/tasks/xchainSetup.ts

```diff
@@ -0,0 +1,206 @@
+import { task } from "hardhat/config";
+import { Contract } from "web3-eth-contract";
+import { CombinedHRE } from "./types";
+const assert = require("assert");
+
+async function setupHub(hub: Contract, deployer: string, parentMessenger: string, childChainId: number) {
+  const [owner, existingParentMessenger] = await Promise.all([
+    hub.methods.owner().call(),
+    hub.methods.messengers(childChainId).call(),
+  ]);
+
+  assert(owner === deployer, `Accounts[0] (${deployer}) is not equal to hub owner (${owner})`);
+  if (existingParentMessenger !== parentMessenger) {
+    console.log(`Setting hub messenger for ID ${childChainId} to ${parentMessenger}...`);
+    const setMessengerTxn = await hub.methods.setMessenger(childChainId, parentMessenger).send({ from: deployer });
+    console.log(`...txn: ${setMessengerTxn.transactionHash}`);
+  }
+}
+
+task("setup-l1-arbitrum-xchain", "Configures L1 cross chain smart contracts for Arbitrum bridge").setAction(
```

Comment:
How much of this function will be identical to optimism and boba?

Are we planning to generalize this when we start working on those deployments?

Reply:
I think the only difference with arbitrum is that prerequisite step of sending ETH to the messenger. But yes I was planning to generalize after implementing for optimism/boba

Reply:
That sounds good!